### ISSUE TITLE
move batchnorm contiguous checks into C

### ISSUE THREAD

**Issue by soumith** [pytorch/pytorch#603]

**Comment by alykhantejani** [comment]
@soumith this only seems to be a problem for the cudnn version if I understand correctly?

I can send a PR if no one is working on this. 

I think I just need to add `THVoidTensor_assertContiguous(input);` to the `cudnn_batch_norm_forward` function and a 

```C++
THVoidTensor_assertContiguous(input); 
THVoidTensor_assertContiguous(grad_output)
```
to `cudnn_batch_norm_backward`

**Comment by soumith** [comment]
yea since this task was created the stuff has been moved into C.

Similar to https://github.com/pytorch/pytorch/blob/master/torch/csrc/autograd/functions/batch_normalization.cpp#L150 you need to add the .contiguous() to the cudnn and other codepaths.
